src/storage/src/hummock/iterator/merge_inner.rs

```diff
@@ -144,7 +158,21 @@ pub type UnorderedMergeIteratorInner<I: HummockIterator> =
     MergeIteratorInner<I, UnorderedNodeExtra>;
 
 impl<I: HummockIterator> UnorderedMergeIteratorInner<I> {
-    pub fn new(iterators: impl IntoIterator<Item = I>, stats: Arc<StateStoreMetrics>) -> Self {
+    pub fn new(iterators: impl IntoIterator<Item = I>) -> Self {
+        Self::create(iterators, None)
+    }
+
+    pub fn for_compactor(
```

Comment:
Instead of forcing all `HummockIterator` of `Compactor` to pass in a global stats, we may force compactor to collect the metrics of `HummockIterator` and report it.